sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -257,14 +257,16 @@ def invalidate_environment(self, name: str) -> None:
         )
 
     @transactional()
-    def delete_expired_snapshots(self) -> t.List[SnapshotTableCleanupTask]:
+    def delete_expired_snapshots(
+        self, ignore_ttl: bool = False
+    ) -> t.List[SnapshotTableCleanupTask]:
         current_ts = now_timestamp(minute_floor=False)
 
-        expired_query = (
-            exp.select("name", "identifier", "version")
-            .from_(self.snapshots_table)
-            .where(exp.column("expiration_ts") <= current_ts)
-        )
+        expired_query = exp.select("name", "identifier", "version").from_(self.snapshots_table)
```

Comment:
There will be a significant runtime penalty

Reply:
Only if the user specifies `--ignore-ttl`, which means that they do want every snapshot checked. I think if theyre going down this road then they can accept the penalty